src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultState = { ...state };
+  const result = [];
+
+  for (const key of actions) {
```

Comment:
The variable 'key' is not descriptive of the actual content it holds. Since it represents an action, a more descriptive name such as 'action' would improve code readability.